src/content_manager.cc

```diff
@@ -496,6 +496,19 @@ int ContentManager::_addFile(const fs::path& path, fs::path rootPath, bool recur
     if (recursive && IS_CDS_CONTAINER(obj->getObjectType())) {
         addRecursive(path, hidden, task);
     }
+
+    if (obj->hasResource(CH_RESOURCE)) {
+        std::string newPath = path.parent_path().c_str();
+        int objectID = storage->findObjectIDByPath(newPath, false);
+        if (objectID != INVALID_OBJECT_ID) {
+            // as there is no proper way to force refresh of unchanged files, delete whole dir and rescan it
+            removeObject(objectID);
+            auto adir = getAutoscanDirectory(objectID);
```

Comment:
If object is removed (I assume from the DB) what does getAutoscanDirectory return? Path is gone or?

Another thing is unclear: what prevents recursion? I.e. if folder.jpg is scanned and then triggers removal of directory...

At least this is not crystal clear from the code.

Reply:
getAutoscan is a leftover, will remove.

Adding a new directory uses another entry point into content manager, as far as I remember.

Reply:
So where will it readd files from the (just removed) directory?

Reply:
Could you call `_addFile()` and it should refresh it?

Reply:
No, as you pointed out the file doesnt change.

I guess we need to enhance the storage layer to be able to update resources.

Reply:
Probably a directory should be a unit of scan. This way it would be easier to keep track of such compound objects: multiple sound and subtitle tracks (for different languages) as well as case-insensitive file names for folder.jpg (if really important).

This makes sense only for directories with small number of files (say, 100, maybe configurable).